Library/Homebrew/extend/ENV/std.rb

```diff
@@ -293,7 +293,7 @@ def cxx11
     if compiler == :clang
       append "CXX", "-std=c++11"
       append "CXX", "-stdlib=libc++"
-    elsif compiler =~ /gcc-(4\.(8|9)|5)/
+    elsif compiler =~ /^gcc-(4\.[8-9]|[5-8]|14)$/
```

Comment:
Maybe here and below `CompilerConstants.GNU_GCC_REGEXP` can be used? That way a version can be added in one place/file

Reply:
After all the testing, I'd prefer it if we could land this as-is, and then deal with the infrastructure changes separately afterwards.
Upstream dealt with this issue already & we can bring that back for consistency, as a separate change, after.